impeller/entity/geometry/stroke_path_geometry.cc

```diff
@@ -38,225 +82,312 @@ Join StrokePathGeometry::GetStrokeJoin() const {
-        Point orientation = offset * (reverse ? -1 : 1);
-
-        VS::PerVertexData vtx;
-
-        Point forward(offset.y, -offset.x);
-        Point forward_normal = forward.Normalize();
-
-        CubicPathComponent arc;
-        if (reverse) {
-          arc = CubicPathComponent(
-              forward,
-              forward + orientation * PathBuilder::kArcApproximationMagic,
-              orientation + forward * PathBuilder::kArcApproximationMagic,
-              orientation);
-        } else {
-          arc = CubicPathComponent(
-              orientation,
-              orientation + forward * PathBuilder::kArcApproximationMagic,
-              forward + orientation * PathBuilder::kArcApproximationMagic,
-              forward);
-        }
-
-        vtx.position = position + orientation;
-        vtx_builder.AppendVertex(vtx);
-        vtx.position = position - orientation;
-        vtx_builder.AppendVertex(vtx);
-        std::vector<Point> arc_points;
-        arc.AppendPolylinePoints(scale, arc_points);
-        for (const auto& point : arc_points) {
-          vtx.position = position + point;
-          vtx_builder.AppendVertex(vtx);
-          vtx.position = position + (-point).Reflect(forward_normal);
-          vtx_builder.AppendVertex(vtx);
-        }
-      };
-      break;
+      return &StrokePathGeometry::CreateRoundCap;
     case Cap::kSquare:
-      cap_proc = [](VertexBufferBuilder<VS::PerVertexData>& vtx_builder,
-                    const Point& position, const Point& offset, Scalar scale,
-                    bool reverse) {
-        Point orientation = offset * (reverse ? -1 : 1);
-
-        VS::PerVertexData vtx;
-
-        Point forward(offset.y, -offset.x);
-
-        vtx.position = position + orientation;
-        vtx_builder.AppendVertex(vtx);
-        vtx.position = position - orientation;
-        vtx_builder.AppendVertex(vtx);
-        vtx.position = position + orientation + forward;
-        vtx_builder.AppendVertex(vtx);
-        vtx.position = position - orientation + forward;
-        vtx_builder.AppendVertex(vtx);
-      };
-      break;
+      return &StrokePathGeometry::CreateSquareCap;
   }
-  return cap_proc;
 }
 
-// static
-VertexBufferBuilder<SolidFillVertexShader::PerVertexData>
-StrokePathGeometry::CreateSolidStrokeVertices(
-    const Path& path,
-    Scalar stroke_width,
-    Scalar scaled_miter_limit,
-    const StrokePathGeometry::JoinProc& join_proc,
-    const StrokePathGeometry::CapProc& cap_proc,
-    Scalar scale) {
-  VertexBufferBuilder<VS::PerVertexData> vtx_builder;
-  auto point_buffer = std::make_unique<std::vector<Point>>();
-  // 512 is an arbitrary choice that should be big enough for most paths without
-  // needing to reallocate. If we have motivating benchmarks we should raise or
-  // lower this number, cause dnfield just made it up!
-  point_buffer->reserve(512);
-  auto polyline = path.CreatePolyline(scale, std::move(point_buffer));
+struct StrokeGenerator {
```

Comment:
This should be a class https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes

There are invariants like `offset`.

Reply:
Done